### PR TITLE
Hw2

### PR DESCRIPTION
**Pull request by KristinaBeslic**

### CODE REVIEW

File: app/src/main/java/agency/five/codebase/android/movieapp/ui/home/HomeScreen.kt [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+
+    HomeScreen(
+        popular = popularCategoryViewState,
+        nowPlaying = nowPlayingCategoryViewState,
+        upcoming = upcomingCategoryViewState,
+        modifier = Modifier,
+        onNavigateToMovieDetails = onNavigateToMovieDetails,
+        onCategoryClick = { categoryId ->
+            when(categoryId) {
+                0, 1, 2, 3 -> {
```
Comment (ghost) [inline app/src/main/java/agency/five/codebase/android/movieapp/ui/home/HomeScreen.kt]: Use `.ordinal` properties of enums to resolve categoryId. 
Such as:
```kotlin
MovieCategory.POPULAR_STREAMING.ordinal
```

File: app/src/main/java/agency/five/codebase/android/movieapp/ui/main/MainScreen.kt [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+    val navController = rememberNavController()
+    val navBackStackEntry by navController.currentBackStackEntryAsState()
+    var showBottomBar by remember { mutableStateOf(true) }
+    var showBackIcon = !showBottomBar
+    Scaffold(
+        topBar = {
+            TopBar(
+                navigationIcon = {
+                    if (showBackIcon) {
+                        navController.navigateUp()
```
Comment (ghost) [inline app/src/main/java/agency/five/codebase/android/movieapp/ui/main/MainScreen.kt]: You don't need this

File: app/src/main/java/agency/five/codebase/android/movieapp/ui/main/MainScreen.kt [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+                        onNavigateToMovieDetails = { movieId ->
+                            navController.navigate(
+                                NavigationItem.MovieDetailsDestination.createNavigationRoute(movieId)
+                            )
+                            showBottomBar = false
+                        }
+                    )
+                }
+                composable(NavigationItem.FavoritesDestination.route) {
+                    showBottomBar = true
```
Comment (ghost) [inline app/src/main/java/agency/five/codebase/android/movieapp/ui/main/MainScreen.kt]: This solution doesn’t scale well. Imagine that you need to add more screens, than you’ll have to put the logic for showing bottomBar/backIcon into every screen:
```kotlin
composable(someRoute) {
  showBottomBar = true/false
  showBackIcon = false/true
}
```
You can use provided navBackStackEntry for calculating showBottomBar:
```kotlin
val showBottomBar = when (navBackStackEntry?.destination?.route) {
    MovieDetailsDestination.route -> false
    else -> true
}

val showBackIcon = !showBottomBar
```

File: app/src/main/java/agency/five/codebase/android/movieapp/ui/main/MainScreen.kt [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+                destination = destination,
+                currentDestination = currentDestination,
+                onNavigateToDestination = onNavigateToDestination
+            )
+        }
+    }
+}
+
+@Composable
+fun RowScope.AddItem(
```
Comment (ghost) [inline app/src/main/java/agency/five/codebase/android/movieapp/ui/main/MainScreen.kt]: Should just be called `Item` or `BottomNavItem` or something similar.
Composable functions represent a UI component and not an action to be performed (AddItem -> Item).

**Review by ghost — CHANGES_REQUESTED** [review]

**Comment by ghost** [comment]
@KristinaBeslic 
> On card click navigation doesn't work properly, top billed cast on movie details screen doesn't show properly

Please schedule consultations with a member of our team to review this in more detail

**Comment by KristinaBeslic** [comment]
> @KristinaBeslic
> 
> > On card click navigation doesn't work properly, top billed cast on movie details screen doesn't show properly
> 
> Please schedule consultations with a member of our team to review this in more detail

Solved :)

**Review by ghost — APPROVED** [review]
🚀

**Pull request merged**